funsor/torch/distributions.py

```diff
@@ -137,7 +137,7 @@ def _infer_value_domain(**kwargs):
 @functools.lru_cache(maxsize=5000)
 def _infer_value_domain(cls, **kwargs):
     instance = cls.dist_class(**{k: dummy_numeric_array(domain) for k, domain in kwargs.items()}, validate_args=False)
-    return Reals[instance.event_shape]
+    return Reals[instance.batch_shape + instance.event_shape]
```

Comment:
Hmm can you explain this? As I understood only `backend.Independent` distributions would trigger batch dims being converted to event dims, but this line suggests we no longer support default interpretation of batch_shape as funsor inputs 😕

Reply:
`_infer_value_domain` takes the `.output` domains of the distribution's parameters as arguments, so `instance.batch_shape` contains only *output* dimensions of the parameters/value, not dimensions that would have otherwise been converted to funsor inputs.

Before this PR, passing a distribution parameters with extra output dimensions would simply error; now these dimensions broadcast:
```py
with interpretation(lazy):
    d = funsor.torch.distributions.Normal(loc=funsor.Tensor(torch.zeros(3), OrderedDict()), scale=1.)
assert d.inputs["value"] == d.loc.output == Reals[3]
```

Reply:
Thanks for explaining!